test/cloudtest/test_recovery.py

```diff
@@ -0,0 +1,252 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+import subprocess
+from dataclasses import dataclass
+from textwrap import dedent
+
+import pytest
+
+from materialize.cloudtest.app.materialize_application import MaterializeApplication
+
+
+@dataclass
+class ClusterSetup:
+    testcase_index: int
+    storage_replication_factor: int = 1
+    compute_replication_factor: int = 1
+
+    def get_storage_name(self) -> str:
+        return f"storage_t{self.testcase_index}"
+
+    def get_compute_name(self) -> str:
+        return f"compute_t{self.testcase_index}"
+
+    def get_counter_source_name(self) -> str:
+        return f"counter_{self.testcase_index}"
+
+    def get_mv_name(self) -> str:
+        return f"mv_{self.testcase_index}"
+
+
+def populate(mz: MaterializeApplication, cluster_setup: ClusterSetup) -> None:
+    mz.environmentd.sql(
```

Comment:
I believe `sql` can take multiple SQL statements separated by `;` so consider using `dedent` and a single `"""` multi-line block for all of this SQL.

Reply:
Good idea 👍